registrations/permissions.py

```diff
@@ -51,6 +51,30 @@ def has_object_permission(
         return obj.can_be_edited_by(request.user)
 
 
+class CanAccessRegistrationSignups(
+    UserDataFromRequestMixin, permissions.BasePermission
+):
+    def has_permission(self, request: Request, view: APIView) -> bool:
```

Comment:
Trivial note: `view: APIView` would benefit from a more specific type. The permissions.py has the following mypy errors for APIView:

error: "APIView" has no attribute "action"  [attr-defined]
error: "APIView" has no attribute "user_data_source_and_organization"  [attr-defined]

One workaround could be to define a Union type like

```python
from rest_framework.viewsets import GenericViewSet
from events.api import UserDataSourceAndOrganizationMixin

GenericRegistrationViewSet: Union[UserDataSourceAndOrganizationMixin, GenericViewSet]
```

But yeah anyway, no changes requested.

Reply:
My suggestion apparently does catch fire in practice `ImportError: cannot import name 'UserDataSourceAndOrganizationMixin' from partially initialized module 'events.api'`